### ISSUE TITLE
Rescue script (load firmware while booting with a btn pressed)

### ISSUE THREAD

**Issue by spiccinini** [LibreRouterOrg/u-boot#3]
Provide a way to load a new firmware using a _rescue script_ that is trigered when booting while pressing the button. Ideally it is an http server like in https://github.com/pepe2k/u-boot_mod  but a simpler tftp command could be used.
Another idea is to enable netconsole when the btn is pressed.

**Comment by CodeFetch** [comment]
A router manufacturer is going to sell some routers without firmware (only bootloader). Thus I'm working on a bootloader that is easier to use including a DHCP server for the web interface, a mode where it acquires a DHCP lease and tries to load an image with a board specific filename from the gateway IP address and one where it loads an image using TFTP using a static IP address compatible to the stock U-Boots that support TFTP loading (e.g. TP-Link WR841N v9). The mode depends on how long you press the reset button. At the end the netconsole is being activated. I don't like the ordinary bootp flashing approaches for routers, that why I do things a bit different. Most of it is already working in my local tree. I'm going to inform you when I've pushed it.

The DHCPD is based on [this implementation](https://github.com/Mleaf/OpBoot-mt7688/blob/master/uip/apps/dhcpd/dhcpd.c) using the uIP stack like pepe2k's httpd.

**Comment by spiccinini** [comment]
@CodeFetch this is excellent!!! Please let me know if you need a hand! We can also enter this mode directly if no firmware can't be booted (because invalid checksum or there is no firmware like from your manufacturer)

**Comment by CodeFetch** [comment]
Exactly, when the firmware can't be loaded, the web interface is started... That's the plan.
